c10/macros/Macros.h

```diff
@@ -332,7 +332,9 @@ constexpr uint32_t CUDA_THREADS_PER_BLOCK_FALLBACK = 256;
 // CUDA_KERNEL_ASSERT checks the assertion
 // even when NDEBUG is defined. This is useful for important assertions in CUDA
 // code that would otherwise be suppressed when building Release.
-#if defined(__ANDROID__) || defined(__APPLE__) || defined(USE_ROCM)
+#if defined(__ANDROID__) || defined(__APPLE__) ||  \
+    (defined(USE_ROCM) && ROCM_VERSION < 40100) || \
+    (defined(USE_ROCM) && defined(ROCM_DISABLE_GPU_ASSERTS))
```

Comment:
Hmm, can't we just replace this one with `|| defined(ROCM_DISABLE_GPU_ASSERTS)` and define it in CMake if version is less than 40100?

Reply:
This is a new build argument to disable asserts in ROCm > 4.1. This is done due to performance regression when asserts are enabled for ROCm.